injected/src/features/autofill-password-import.js

```diff
@@ -77,38 +102,164 @@ export default class AutofillPasswordImport extends ContentFeature {
                       style: this.signInButtonStyle,
                       element,
                       shouldTap: this.#signInButtonSettings?.shouldAutotap ?? false,
+                      shouldWatchForRemoval: false,
                   }
                 : null;
         } else {
             return null;
         }
     }
 
+    /**
+     * Removes the overlay if it exists.
+     */
+    removeOverlayIfNeeded() {
+        const existingOverlay = document.getElementById(OVERLAY_ID);
+        if (existingOverlay != null) {
+            existingOverlay.style.display = 'none';
+            existingOverlay.remove();
+            document.removeEventListener('scroll', this.updateOverlayPosition.bind(this));
```

Comment:
This won't remove the other listener as there is a closure involved (but it wouldn't be equivelant even if you added it here and also the bind produces a completely new function instance too).

So there's wither two ways:

1. Store the scroll listener in a global:

```js
this.scrollListener = () => {...}
document.addEventListener('scroll', this.scrollListener);
....
document.removeEventListener('scroll', this.scrollListener);
```

2. Class based event handler:
```js
export default class AutofillPasswordImport extends ContentFeature {
    insertOverlay() {
        document.addEventListener('scroll', this);
    }
    removeOverlayIfNeeded() {
        document.removeEventListener('scroll', this);
    }
    handleEvent(e) {
        if (e.type === 'scroll') {
            requestAnimationFrame(() => this.updateOverlayPosition(overlay, style));
        }
    }
}
```

My preferred is 2. purely as it doesn't require another global and also centralises event handling.